File: src/codeforces/round578Div2/Hotelier.java
package codeforces.round578Div2;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Hotelier {

    private static int[] hotel = new int[10];

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        String events = br.readLine();

        int l = 0;
        int r = 9;

        for (int i = 0; i < n; i++) {
            if (events.charAt(i) == 'L'){

                boolean changed1 = false;
                for (int j = 0; j < l && !changed1; j++) {
                    if (hotel[j] == 0){
                        hotel[j] = 1;
                        changed1 = true;
                    }

                }
                if (!changed1){
                    hotel[l] = 1;
                    l ++;
                }

            }else if (events.charAt(i) == 'R'){

                boolean changed2 = false;
                for (int j = 9; j > r && !changed2; j--) {
                    if (hotel[j] == 0){
                        hotel[j] = 1;
                        changed2 = true;
                    }
                }

                if (!changed2){
                    hotel[r] = 1;
                    r--;
                }

            }else {
                int room = Integer.parseInt(events.charAt(i) + "");
                hotel[room] = 0;
            }
        }

        for (int h : hotel) {
            System.out.print(h);
        }
    }
}

File: src/codeforces/div2/round578Div2/Hotelier.kt
package codeforces.div2.round578Div2

import java.io.BufferedReader
import java.io.InputStreamReader

val hotel = Array<Int>(10){x -> 0}

fun main(args: Array<String>) {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val n = br.readLine().toInt()
    val events = br.readLine()

    var l = 0
    var r = 9

    for (i in 0 until n){
        if (events[i] == 'L'){

            var changed = false
            for (j in 0 .. l){
                if (hotel[j] == 0){
                    l = j
                    changed = true
                    break
                }
            }

            if (!changed){
                hotel[l] = 1
                l ++
            }

        }
        else if (events[i] == 'R'){

            var changed = false
            for (j in 9 .. r){
                if (hotel[j] == 0){
                    r = j
                    changed = true
                    break
                }
            }

            if (!changed){
                hotel[r] = 1
                r --
            }
        } else {
            val room = events[i].toString().toInt()
            hotel[room] = 0
        }
    }

    for (i in 0 .. 9){
        print(hotel[i])
    }
}